polaris-core/src/main/java/org/apache/polaris/core/persistence/cache/EntityCache.java

```diff
@@ -34,6 +36,7 @@
 import org.jetbrains.annotations.Nullable;
 
 /** The entity cache, can be private or shared */
+@RealmScope
```

Comment:
IMO a cache should leverage the available/configured resources of the JVM in a way that the resources are used most efficiently but also cannot exceed the available heap.

Here the cache is scoped to a realm, where each realm (number of realms is unbounded) permanently allocates JVM resources.

Not related to this PR, but something that should be fixed.

Reply:
That's a fine callout and I think revisiting the EntityCache to be realm-aware, rather than distinct per-realm, is a good change to make, but agreed that it shouldn't be done here